File: SchoopyAdmin/src/pkgData/Person.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pkgData;

import com.jfoenix.controls.datamodels.treetable.RecursiveTreeObject;
import java.util.Objects;

/**
 *
 * @author schueler
 */
public abstract class Person {
    String firstName, lastName;
    String schoolemail;
    String username; //for teachers its kürzel for schueler its lastname+first letter of firstname
    String password;  

    public Person(){
    
    }
    
    public Person(String firstName, String lastName, String schoolemail, String username, String password) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.schoolemail = schoolemail;
        this.username = username;
        this.password = password;
    }
    public Person(String firstName, String lastName, String schoolemail, String username) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.schoolemail = schoolemail;
        this.username = username;
    }

    
    
    
    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getSchoolemail() {
        return schoolemail;
    }

    public void setSchoolemail(String schoolemail) {
        this.schoolemail = schoolemail;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 43 * hash + Objects.hashCode(this.username);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Person other = (Person) obj;
        if (!Objects.equals(this.username, other.username)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return firstName + " " + lastName + ", schoolemail=" + schoolemail + ", username=" + username;
    }
    
    
    
}

File: SchoopyC#/Schoopy/Data/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Schoopy
{
    public abstract class Person
    {
        public string firstName { get; set; }
        public string lastName { get; set; }
        public string schoolemail { get; set; }
        public string username { get; set; }
        public string password { get; set; }

        public Person(string _username, string _password)
        {
            this.username = _username;
            this.password = _password;
        }

        public Person(string _username, string _password , string _firstname, string lastname, string schoolemail)
        {
            this.username = _username;
            this.password = _password;
            this.firstName = _firstname;
            this.lastName = lastName;
            this.schoolemail = schoolemail;
        }

        public Person()
        {

        }
    }
}
